server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java

```diff
@@ -137,18 +159,26 @@ private void buildMimeHeaders(Message.Builder messageBuilder, CreationMessageId
 
         // note that date conversion probably lose milliseconds!
         messageBuilder.setDate(Date.from(newMessage.getDate().toInstant()), TimeZone.getTimeZone(newMessage.getDate().getZone()));
-        newMessage.getInReplyToMessageId().ifPresent(addInReplyToHeader(messageBuilder::addField));
+        newMessage.getInReplyToMessageId()
+            .ifPresent(id -> addHeader(messageBuilder, IN_REPLY_TO_HEADER, id));
         if (!isMultipart(newMessage, messageAttachments)) {
             newMessage.getHtmlBody().ifPresent(x -> messageBuilder.setContentType(HTML_MEDIA_TYPE, UTF_8_CHARSET));
         }
+        newMessage.getHeaders().entrySet().stream()
+            .filter(header -> ! header.getKey().trim().isEmpty())
```

Comment:
if you need a trim here, maybe you should apply it in a map step for following steps to benefit from it, don't you think ?

Reply:
no I don't need it after because mime4j handles it. I just need it to be sure to not add an empty header from a space only header